bluemira/equilibria/equilibrium.py

```diff
@@ -1332,3 +1341,318 @@ def plot_core(self):
         Plot a 1-D section through the magnetic axis.
         """
         return CorePlotter2(self)
+
+
+class SHApproximation(Equilibrium):
```

Comment:
Can we have a meeting to discuss the architecture here? I don't see the need for a separate class. I was expecting spherical harmonics to be integrated in terms of constraints and supporting functions.

If we need something to return an SH approximation of an Equilibrium from some Equilibrium, this feels more like a function which takes in an Equilibrium and returns and Equilibrium.. but perhaps I'm missing something? Keen to discuss :)

Reply:
I agree I was full review mode and completely glossed over that, I'll setup a meeting

Reply:
I have made the architecture changes, as discussed in the meeting yesterday, going to resolve this conversation for now.